src/osm2pgsql-gen.cpp

```diff
@@ -0,0 +1,365 @@
+Generalizes OSM data.
+
+Use PG* environment variables to specify database connection.
+Without x,y,X,Y the extent will be read from the database table.
+
+Options:
+    -h, --help                    Print this help text and stop
+    -b, --buffer=SIZE             Buffer size
+    -D, --no-delete               Do not DELETE existing features
+    -e, --expire-list=TABLE       Read tiles to render from database table
+    -g, --geom-column=COLUMN      Geometry column (default: 'geom')
+    -G, --group-by-column=COLUMN  Column name to group table contents on
+    -l, --log-level=LEVEL         Log level (debug, info (default), warn, error)
+    -m, --margin=MARGIN           Margin as a fraction of the tile size
+    -p, --param=KEY=VALUE         Add parameter for strategy (can be used multiple times)
+    -s, --strategy=STRATEGY       Simplification strategy
+    -t, --table=TABLE             Table that should be simplified
+    -T, --dest-table=TABLE        Destination table (must have geom, x, y)
+    -w, --width=SIZE              Size (width/height) of the rendered image (default: 2048)
+    -x                            Minimum tile X
+    -y                            Minimum tile Y
+    -X                            Maximum tile X
+    -Y                            Maximum tile Y
+    -z, --zoom=ZOOM               Zoom level
+    --img-path=PATH               Dump rendered images to PATH
+    --img-table=TABLE             Dump rendered images to TABLE
+    --log-sql                     Log SQL commands
+)");
+}
+
+static char const *const short_options = "hb:De:g:G:l:m:p:s:t:T:w:x:y:X:Y:z:";
+
+static std::array<option, 18> const long_options = {
+    {{"help", no_argument, nullptr, 'h'},
+     {"buffer", required_argument, nullptr, 'b'},
+     {"no-delete", no_argument, nullptr, 'D'},
+     {"expire-list", required_argument, nullptr, 'e'},
+     {"geom-column", required_argument, nullptr, 'g'},
+     {"group-by-column", required_argument, nullptr, 'G'},
+     {"log-level", required_argument, nullptr, 'l'},
+     {"margin", required_argument, nullptr, 'm'},
+     {"param", required_argument, nullptr, 'p'},
+     {"strategy", required_argument, nullptr, 's'},
+     {"table", required_argument, nullptr, 't'},
+     {"dest-table", required_argument, nullptr, 'T'},
+     {"width", required_argument, nullptr, 'w'},
+     {"zoom", required_argument, nullptr, 'z'},
+     {"img-path", required_argument, nullptr, 200},
+     {"img-table", required_argument, nullptr, 201},
+     {"log-sql", no_argument, nullptr, 202},
+     {nullptr, 0, nullptr, 0}}};
+
+static std::array<uint32_t, 4> get_extent_from_db(pg_conn_t *connection,
+                                                  std::string const &table,
+                                                  uint32_t zoom)
+{
+    connection->query(
+        PGRES_COMMAND_OK,
+        "PREPARE get_extent AS WITH ex AS "
+        "(SELECT ST_EstimatedExtent('{}', 'geom') AS e) "
```

Comment:
This hard-codes the geometry column to `geom`

Reply:
fixes in new version